sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -46,6 +46,10 @@ trait Catalog {
 
   def lookupRelation(tableIdent: TableIdentifier, alias: Option[String] = None): LogicalPlan
 
+  def setCurrentDatabase(databaseName: String): Unit = {
+    throw new UnsupportedOperationException
```

Comment:
Why create a dummy implementation here? This is not consistent with the way Catalog is currently written. The other option creates more overhead though.